File: apps/practice_app/practice_app/realtime/handlers.py
print("🔥 REALTIME HANDLERS LOADED")
from frappe.realtime import Socket, realtime
import frappe
@realtime.on("project_subscribe")
def project_subscribe(socket: Socket, project: str):

    print("Socket ID:", socket.id)
    print("Before:", socket.rooms)

    if socket.has_permission("Project", project):
        socket.join(f"project:{project}")

    print("After:", socket.rooms)

File: apps/practice_app/practice_app/realtime/handlers.js
function practice_app_handlers(socket) {

    console.log("🔥 PRACTICE APP HANDLERS LOADED");

    socket.on("project_subscribe", (project) => {

        console.log("🔥 EVENT RECEIVED");
        console.log("Project:", project);
        console.log("Socket ID:", socket.id);

    });
}

module.exports = practice_app_handlers;